walkbox.go

```diff
@@ -80,24 +83,49 @@ func (w *WalkBox) ContainsPoint(p *Positionf) bool {
 	return numberOfIntersections%2 == 1 // Odd count means inside
 }
 
-// IsAdjacent checks if two WalkBoxes are adjacent. It returns false if either WalkBox is disabled.
-func (w *WalkBox) IsAdjacent(otherWalkBox *WalkBox) bool {
+// isAdjacent checks if two WalkBoxes are adjacent. It returns false if either WalkBox is disabled.
+func (w *WalkBox) isAdjacent(otherWalkBox *WalkBox) bool {
+	return w.gateWith(otherWalkBox) != nil
+}
+
+// gateWith returns the gate (shared point) between two adjacent walk boxes, if any.
+// It returns nil if either WalkBox is disabled.
+func (w *WalkBox) gateWith(otherWalkBox *WalkBox) *Positionf {
 	if w.enabled && otherWalkBox.enabled {
 		for _, vertex := range otherWalkBox.vertices {
-			if w.ContainsPoint(vertex) {
-				return true
+			if w.containsPoint(vertex) {
+				return vertex
```

Comment:
I have doubts about this implementation. 

I guess the gate point will be used in the future to determine the path across boxes. So we calculate the itinerary first, and then we map it into points using the gates as intermediate positions. 

However, as we are calculating the gates like that, we will have paths with this form:
![image](https://github.com/user-attachments/assets/580024cb-dbc3-4be1-aa1b-c3bf511da7dc)

Going from A to B, we will have to traverse the gate, that will be calculated as the first common vertex. In this case, the top-right one. Having a very weird path. 

Is that correct?

Reply:
Considering that we have `ClosestPointOnSegment()` function, why don't we use it? We could obtain the pair of vertices that comprise the gate and calculate the shortest point of that segment from the previous waypoint.

Reply:
You are right, this approach is soo naive. Let me take a look, and add little more unit tests ok?
(Im gonna exceed the 200 line limit but is for a good reason)

It is an amaizing idea!